### PR TITLE
front prestop hook 

### PR DESCRIPTION
**Pull request by fontanierh**
- front-edge only for now to test
- introduces "wakeLock" which allows to protect a piece of code from shutdowns
- the k8s "preStop" hook calls a (protected) endpoint on the app that hangs until there are no wakeLocks

### CODE REVIEW

File: k8s/deployments/front-edge-deployment.yaml [hunk 1]
```diff
@@ -16,6 +16,7 @@ spec:
       annotations:
         ad.datadoghq.com/web.logs: '[{"source": "front","service": "front-edge","tags": ["env:prod"]}]'
     spec:
+      terminationGracePeriodSeconds: 60
```
Comment (fontanierh) [inline k8s/deployments/front-edge-deployment.yaml:19]: bumped to 1 minute from 30 seconds

File: front/lib/api/assistant/pubsub.ts [hunk 2]
```diff
@@ -96,7 +97,7 @@ async function handleUserMessageEvents(
 ): Promise<Result<UserMessageType, PubSubError>> {
   const promise: Promise<Result<UserMessageType, PubSubError>> = new Promise(
     (resolve) => {
-      void (async () => {
+      void wakeLock(async () => {
```
Comment (spolu) [inline front/lib/api/assistant/pubsub.ts:100]: You're likely missing the `retryAgentMessageWithPubSub` one?

Comment (fontanierh) [reply front/lib/api/assistant/pubsub.ts:100]: Yep and the cancel I guess ?

Comment (spolu) [reply front/lib/api/assistant/pubsub.ts:100]: Otherwise LGTM!

Comment (spolu) [reply front/lib/api/assistant/pubsub.ts:100]: Well cancel is not waiting on anything

**Review by lasryaric — CHANGES_REQUESTED** [review]
Mini changes.

**Pull request merged**